webapp/client/src/pages/TaxNumberPage.js

```diff
@@ -0,0 +1,331 @@
+          exampleInput: t("lotseFlow.taxNumber.taxNumberInput.labelText"),
+        }}
+        errors={fields.steuernummer.errors}
+        key={selectedStateAbbreviation} // Added render a new DOM Element whenever the selected state changes
+        isSplit={selectedStateAbbreviation !== "he"} // Do not split field for Hessen
+      />
+    </FormRowCentered>
+  );
+
+  const requestNewTaxNumberCheckbox = (
+    <FormRowCentered key="requestNewTaxNumberRow">
+      <SubHeading>
+        {t("lotseFlow.taxNumber.requestNewTaxNumber.headline")}
+      </SubHeading>
+      <div>{t("lotseFlow.taxNumber.requestNewTaxNumber.intro")}</div>
+      <FormFieldConsentBox
+        required
+        fieldName="request_new_tax_number"
+        fieldId="request_new_tax_number"
+        checked={fields.requestNewTaxNumber.checked}
+        labelText={
+          <Trans
+            t={t}
+            i18nKey="lotseFlow.taxNumber.requestNewTaxNumber.labelText"
+            count={numberOfUsers}
+          />
+        }
+        errors={fields.requestNewTaxNumber.errors}
+      />
+    </FormRowCentered>
+  );
+
+  let shownFields;
+
+  switch (currentState(taxNumberFormState, fields)) {
+    case TAX_NUMBER_FORM_STATES.NothingSelected:
+      break;
+    case TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists:
+      shownFields = [stateDropDown];
+      break;
+    case TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists:
+      shownFields = [stateDropDown];
+      break;
+    case TAX_NUMBER_FORM_STATES.TaxOfficeSelection:
+      shownFields = [stateDropDown, bufaNrDropdown];
+      break;
+    case TAX_NUMBER_FORM_STATES.TaxNumberInput:
+      shownFields = [stateDropDown, taxNumberInput];
+      break;
+    case TAX_NUMBER_FORM_STATES.RequestNewTaxNumber:
+      shownFields = [
+        stateDropDown,
+        bufaNrDropdown,
+        requestNewTaxNumberCheckbox,
+      ];
+      break;
+    default:
+  }
+
+  return (
+    <>
+      <StepHeaderButtons />
+      <FormHeader {...stepHeader} />
+      <StepForm {...form}>
+        <FormRowCentered>
+          <FormFieldYesNo
+            fieldName="steuernummer_exists"
+            fieldId="steuernummer_exists"
+            value={fields.steuernummerExists.value}
+            label={{
+              text: t("lotseFlow.taxNumber.taxNumberExists.labelText", {
+                count: numberOfUsers,
+              }),
+            }}
+            details={{
+              title: t("lotseFlow.taxNumber.taxNumberExists.help.title"),
+              text: t("lotseFlow.taxNumber.taxNumberExists.help.text"),
+            }}
+            errors={fields.steuernummerExists.errors}
+            onChangeHandler={changeTaxNumberExists}
+          />
+        </FormRowCentered>
+        {shownFields}
+      </StepForm>
+    </>
+  );
+}
+
+const fieldPropType = PropTypes.exact({
+  value: PropTypes.any,
+  errors: PropTypes.arrayOf(PropTypes.string),
+});
```

Comment:
Instead of this, you can use the already existing definition in `/lib/propTypes`. Same for `checkboxPropType`.